src/opencl_sha256crypt_fmt_plug.c

```diff
@@ -561,19 +572,25 @@ static int crypt_all(int *pcount, struct db_salt *_salt)
 		int tot_todo = 0, len;
 
 		if (lens_differ) {
-			indices = mem_alloc(gws * sizeof(int));
+
+			if (gws < previous_gws) {
+				previous_gws = gws;
+				MEM_FREE(indices);
+				indices = mem_alloc(gws * sizeof(int));
+			}
 
 			for (len = 0; len <= PLAINTEXT_LENGTH; len++) {
-				for (index = 0; index < count; index++) {
-					if (plaintext[index].length == len)
-						indices[tot_todo++] = index;
+				if (bitmap_of_lens & (1 << len))
```

Comment:
```c
if (!(bitmap_of_lens >> len))
	break;
if ((bitmap_of_lens >> len) & 1)
```

This could probably be sped up a tiny bit by first copying `bitmap_of_len` to a temporary loop variable before the loop, then using `>>= 1` on it (then no variable shift count). But since we have variable shift count inside `set_key()` anyway, which executes many more times, the point is moot.